### DISCUSSION TITLE
Sign transactions without having to connect to a node

### DISCUSSION THREAD

**Discussion by kevinam99** [ethereum/web3.js#5953] [Q&A]
I have a backend service that encodes and signs transactions and then sends them to the chain. This service works well when I need to submit the transactions to the same chain at all times.

I have a requirement now where I want to be able to use the same backend service but the transactions only needs to be encoded and signed. The problem, however, is that the transaction is to be submitted to another chain which is an Ethereum fork. 

Is it possible to do this in web3js where the same service connected to one chain (itself an Ethereum fork) but the tx is encoded and signed (but not submitted) for a different chain without having to connect to the different chain's node?

**Comment by lodembeep** [comment]
I would recommend using ethers.js for offline transactions, there you can prepare your transactions, sign them offline as long as you have the private key, and send it later.

Do not forget to change the chainId accordingly.

https://docs.ethers.org/v5/api/signer/#Wallet

**Reply by kevinam99** [reply]
Thanks @lodembeep. This looks good! How to initialise `signer`, though? (From `signer.SignTransaction()`)

**Reply by lodembeep** [reply]
if you want to do it offline, you better look at the wallet docs : 
```
// Signing a transaction
await walletMnemonic.signTransaction(tx)
// '0xf865808080948ba1f109551bd432803012645ac136ddd64dba72880de0b6b3a7640000801ca0918e294306d177ab7bd664f5e141436563854ebe0a3e523b9690b4922bbb52b8a01181612cec9c431c4257a79b8c9f0c980a2c49bb5a0e6ac52949163eeb565dfc'
```

**Reply by kevinam99** [reply]
Got it, thanks. Will try this.
